airlock/static/assets/file_browser/index.js

```diff
@@ -133,8 +133,22 @@ function addCheckboxClickListener() {
 
 // On first load of the page we need to wire up the event listener
 // so that we can respond to checkbox changes.
-document.addEventListener("DOMContentLoaded", addCheckboxClickListener);
+if (document.readyState !== "loading") {
```

Comment:
Do we need this because of the htmx loads? So the document itself is already loaded but the checkbox listener and status have to be called again for htmx-refreshed content?

Reply:
I think it's actually just for the back button case. The event listener is put on an element far enough up the DOM so that any htmx changes within the file browser don't touch it.  Once its enabled initially you therefore don't need to redo it. However, when the back button is clicked you get a whole page refresh so event listeners are disregarded - but you also don't get a DOMContentLoaded event (or at least it has already triggered before this code runs). I think this is because the page is cached by the browser in some way, but I'm not 100%.